src/glfw3.jl

```diff
@@ -269,15 +269,116 @@ end
 Base.show(io::IO, m::Monitor) = write(io, "Monitor($(m.handle == C_NULL ? m.handle : GetMonitorName(m)))")
 
 const WindowHandle = Ptr{Void}
+
 type Window
 	handle::WindowHandle
 	callbacks::Vector{Function}
 end
 Window(handle::WindowHandle) = Window(handle, Vector{Function}(_window_callbacks_len[]))
+#Came from GLWindow.jl/screen.jl
+"""
+Function to create a pure GLFW OpenGL window
+"""
+function Window(
+        name = "GLWindow";
+        resolution = standard_screen_resolution(),
+        debugging = false,
+        major = 3,
+        minor = 3,# this is what GLVisualize needs to offer all features
+        windowhints = standard_window_hints(),
+        contexthints = standard_context_hints(major, minor),
+        visible = true,
+        focus = false,
+        fullscreen = false,
+        monitor = nothing
+    )
+    WindowHint(VISIBLE, visible)
+    WindowHint(FOCUSED, focus)
+    for ch in contexthints
+        WindowHint(ch[1], ch[2])
+    end
+    for wh in windowhints
+        WindowHint(wh[1], wh[2])
+    end
+
+    @static if is_apple()
+        if debugging
+            warn("OpenGL debug message callback not available on osx")
+            debugging = false
+        end
+    end
+
+    WindowHint(OPENGL_DEBUG_CONTEXT, Cint(debugging))
+
+    monitor = if monitor == nothing
+        GetPrimaryMonitor()
+    elseif isa(monitor, Integer)
+        GetMonitors()[monitor]
+    elseif isa(monitor, Monitor)
+        monitor
+    else
+        error("Monitor needs to be nothing, int, or GLFW.Monitor. Found: $monitor")
+    end
+
+    window = CreateWindow(resolution..., String(name))
+
+    if fullscreen
+        SetKeyCallback(window, (_1, button, _2, _3, _4) -> begin
+            button == KEY_ESCAPE && make_windowed!(window)
+        end)
+        make_fullscreen!(window, monitor)
+    end
+
+    MakeContextCurrent(window)
+
+    debugging && glDebugMessageCallbackARB(_openglerrorcallback, C_NULL)
+    window
+end
+
+
 import Base.==; ==(x::Window, y::Window) = (x.handle == y.handle)
 Base.cconvert(::Type{WindowHandle}, window::Window) = window.handle
 Base.cconvert(::Type{Window}, handle::WindowHandle) = ccall( (:glfwGetWindowUserPointer, lib), Ref{Window}, (WindowHandle,), handle)
 Base.hash(window::Window, h::UInt64) = hash(window.handle, h)
+function Base.isopen(window::Window)
+    window.handle == C_NULL && return false
+    !WindowShouldClose(window)
+end
+
+function Base.resize!(x::Window, w::Integer, h::Integer)
+    SetWindowSize(x, w, h)
+end
```

Comment:
Remove this function (for now). The documented description of `resize!` is for collections, so using it here seems a bit off. The alternative (calling `SetWindowSize`) isn't any longer.